spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
That was my initial thought, too, but we might need a similar option on the outbound adapter...

> Other software (namely Logstash's Redis input/output) uses right push/left pop to achieve the same. 

In which case FIFO/LIFO is not really appropriate.